### DISCUSSION TITLE
How to deposit APT coin in to the module balance ?

### DISCUSSION THREAD

**Discussion by Ghonghito** [aptos-labs/aptos-developer-discussions#152] [Questions] [answered]
### Describe your question in detail. For example how do you get all transactions for a user with the TypeScript SDK?

So, I'm trying to create a module where I can deposit APT coin. But, on this line of code: coin::deposit(&mut holder.balance, coin);, I'm encountering this error: 'Incompatible type '&mut Coin<AptosCoin>', expected 'address'.' I understand that it requires an address, but when I tried to use '@box', it worked. But, when I attempt to make a deposit, it deposits into my wallet, which I used to deploy this module.

### What have you tried or looked at?  Or how can we reproduce the error?

```
module box::deposit {
    use std::signer;
    use aptos_framework::coin::{Self, Coin};
    use aptos_framework::aptos_coin::AptosCoin;

    struct CoinHolder has key, store {
        balance: Coin<AptosCoin>,
    }

    fun init_module(account: &signer) {
        let balance = coin::zero<AptosCoin>();
        move_to(account, CoinHolder { balance });
    }

    public entry fun deposit(account: &signer, amount: u64) acquires CoinHolder {
        let holder = borrow_global_mut<CoinHolder>(signer::address_of(account));
        let coin = coin::withdraw<AptosCoin>(account, amount);
        coin::deposit(&mut holder.balance, coin);
    }
}
```

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by 0xjunha** [comment, marked as answer]
First of all, type signature of coin::deposit is
`public fun deposit<CoinType>(account_addr: address, coin: Coin<CoinType>) acquires CoinStore`

Types of what you provided are:
&mut holder.balance: `&mut Coin<CoinType>`, coin: `Coin<CoinType>`
You should provide `address` type for `account_addr` param.

It worked when you provided @box as `account_addr` arg because @box is an alias for the deployer address in this case.

In Aptos Move, when you deploy a smart contract (Move module), it is published under an Account. "Module" itself doesn't hold APT balance. Account holds "Modules" and "Resources", and APT is one of the "Resources". So, basically your Move module is under the account you used when deploying the module. There's no "Smart Contract Account" created to store the module here.

**Comment by 3oloto** [comment]
thank you for information
